### ISSUE TITLE
Can't install with pip on Windows 10

### ISSUE THREAD

**Issue by asweigart** [r0x0r/pywebview#732]
### Specification
- pywebview version: 3.4
- operating system: Windows 10
- web renderer: n/a

### Description  
  
Running `pip install pywebview` results in an error:

```
C:\Users\Al>pip install pywebview
Collecting pywebview
  Downloading pywebview-3.4-py3-none-any.whl (325 kB)
     |████████████████████████████████| 325 kB 6.8 MB/s
Collecting pythonnet
  Downloading pythonnet-2.5.2.tar.gz (1.9 MB)
     |████████████████████████████████| 1.9 MB ...
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Using legacy 'setup.py install' for pythonnet, since package 'wheel' is not installed.
Installing collected packages: pycparser, pythonnet, pywebview
    Running setup.py install for pythonnet ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\al\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-1fqmpfdv\\pythonnet_bbc2e13de01b43a89e4f315cb6e6ae66\\setup.py'"'"'; __file__='"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-1fqmpfdv\\pythonnet_bbc2e13de01b43a89e4f315cb6e6ae66\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Al\AppData\Local\Temp\pip-record-v0nkxqp9\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\al\appdata\local\programs\python\python39\Include\pythonnet'
         cwd: C:\Users\Al\AppData\Local\Temp\pip-install-1fqmpfdv\pythonnet_bbc2e13de01b43a89e4f315cb6e6ae66\
    Complete output (6 lines):
    usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
       or: setup.py --help [cmd1 cmd2 ...]
       or: setup.py --help-commands
       or: setup.py cmd --help

    error: option --single-version-externally-managed not recognized
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\al\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-1fqmpfdv\\pythonnet_bbc2e13de01b43a89e4f315cb6e6ae66\\setup.py'"'"'; __file__='"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-1fqmpfdv\\pythonnet_bbc2e13de01b43a89e4f315cb6e6ae66\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Al\AppData\Local\Temp\pip-record-v0nkxqp9\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\al\appdata\local\programs\python\python39\Include\pythonnet' Check the logs for full command output.
```

The relevant part of the log file seems to indicate a problem installing pythonnet:

```
2021-06-04T14:42:50,230 Using legacy 'setup.py install' for pythonnet, since package 'wheel' is not installed.
2021-06-04T14:42:50,487 Installing collected packages: pythonnet, pywebview
2021-06-04T14:42:50,489   Created temporary directory: C:\Users\Al\AppData\Local\Temp\pip-record-i1o8ggm6
2021-06-04T14:42:50,489     Running command 'c:\users\al\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-2l_gq3nh\\pythonnet_4ca8ba7d16dc4c389a091dce17a00ebd\\setup.py'"'"'; __file__='"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-2l_gq3nh\\pythonnet_4ca8ba7d16dc4c389a091dce17a00ebd\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Al\AppData\Local\Temp\pip-record-i1o8ggm6\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\al\appdata\local\programs\python\python39\Include\pythonnet'
2021-06-04T14:42:50,877     usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
2021-06-04T14:42:50,877        or: setup.py --help [cmd1 cmd2 ...]
2021-06-04T14:42:50,877        or: setup.py --help-commands
2021-06-04T14:42:50,877        or: setup.py cmd --help

2021-06-04T14:42:50,877     error: option --single-version-externally-managed not recognized
2021-06-04T14:42:50,896 ERROR: Command errored out with exit status 1: 'c:\users\al\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-2l_gq3nh\\pythonnet_4ca8ba7d16dc4c389a091dce17a00ebd\\setup.py'"'"'; __file__='"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-2l_gq3nh\\pythonnet_4ca8ba7d16dc4c389a091dce17a00ebd\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Al\AppData\Local\Temp\pip-record-i1o8ggm6\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\al\appdata\local\programs\python\python39\Include\pythonnet' Check the logs for full command output.
2021-06-04T14:42:50,897 Exception information:
2021-06-04T14:42:50,897 Traceback (most recent call last):
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\req\req_install.py", line 825, in install
2021-06-04T14:42:50,897     success = install_legacy(
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\operations\install\legacy.py", line 81, in install
2021-06-04T14:42:50,897     raise LegacyInstallFailure
2021-06-04T14:42:50,897 pip._internal.operations.install.legacy.LegacyInstallFailure
2021-06-04T14:42:50,897 
2021-06-04T14:42:50,897 During handling of the above exception, another exception occurred:
2021-06-04T14:42:50,897 
2021-06-04T14:42:50,897 Traceback (most recent call last):
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\cli\base_command.py", line 180, in _main
2021-06-04T14:42:50,897     status = self.run(options, args)
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\cli\req_command.py", line 204, in wrapper
2021-06-04T14:42:50,897     return func(self, options, args)
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\commands\install.py", line 393, in run
2021-06-04T14:42:50,897     installed = install_given_reqs(
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\req\__init__.py", line 77, in install_given_reqs
2021-06-04T14:42:50,897     requirement.install(
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\req\req_install.py", line 843, in install
2021-06-04T14:42:50,897     six.reraise(*exc.parent)
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_vendor\six.py", line 703, in reraise
2021-06-04T14:42:50,897     raise value
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\operations\install\legacy.py", line 69, in install
2021-06-04T14:42:50,897     runner(
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\utils\subprocess.py", line 274, in runner
2021-06-04T14:42:50,897     call_subprocess(
2021-06-04T14:42:50,897   File "c:\users\al\appdata\local\programs\python\python39\lib\site-packages\pip\_internal\utils\subprocess.py", line 244, in call_subprocess
2021-06-04T14:42:50,897     raise InstallationSubprocessError(proc.returncode, command_desc)
2021-06-04T14:42:50,897 pip._internal.exceptions.InstallationSubprocessError: Command errored out with exit status 1: 'c:\users\al\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-2l_gq3nh\\pythonnet_4ca8ba7d16dc4c389a091dce17a00ebd\\setup.py'"'"'; __file__='"'"'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-install-2l_gq3nh\\pythonnet_4ca8ba7d16dc4c389a091dce17a00ebd\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Al\AppData\Local\Temp\pip-record-i1o8ggm6\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\al\appdata\local\programs\python\python39\Include\pythonnet' Check the logs for full command output.
2021-06-04T14:42:50,955 Removed build tracker: 'C:\\Users\\Al\\AppData\\Local\\Temp\\pip-req-tracker-wxh68qk8'
```
  
### Practicalities

- YES/NO I am willing to work on this issue myself.

no

- YES/NO I am prepared to support this issue financially.

no

**Comment by r0x0r** [comment]
Which Python version are you using? Check with pythonnet what is the latest version supported.

**Comment by sbbosco** [comment]
@asweigart take a look here #700. Install python.net first from wheel available at https://www.lfd.uci.edu/~gohlke/pythonlibs/#pythonnet , then install pywebview.

**Comment by Youngermaster** [comment]
I had the same Issue:
```
PS C:\Users\juanm\GitHub\Youngermaster\Raspberry-Pi-Codes\ZeroW\webview> pip install pywebview
Collecting pywebview
  Using cached pywebview-3.4-py3-none-any.whl (325 kB)
Collecting pythonnet
  Using cached pythonnet-2.5.2.tar.gz (1.9 MB)
Requirement already satisfied: pycparser in c:\users\juanm\appdata\local\programs\python\python39\lib\site-packages (from pythonnet->pywebview) (2.20)
Using legacy 'setup.py install' for pythonnet, since package 'wheel' is not installed.
Installing collected packages: pythonnet, pywebview
    Running setup.py install for pythonnet ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\juanm\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\juanm\\AppData\\Local\\Temp\\pip-install-h7pab39z\\pythonnet_9cb3293501224b288fc1c9685bb36fbe\\setup.py'"'"'; __file__='"'"'C:\\Users\\juanm\\AppData\\Local\\Temp\\pip-install-h7pab39z\\pythonnet_9cb3293501224b288fc1c9685bb36fbe\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\juanm\AppData\Local\Temp\pip-record-gbzseqx0\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\juanm\appdata\local\programs\python\python39\Include\pythonnet'
         cwd: C:\Users\juanm\AppData\Local\Temp\pip-install-h7pab39z\pythonnet_9cb3293501224b288fc1c9685bb36fbe\
    Complete output (6 lines):
    usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
       or: setup.py --help [cmd1 cmd2 ...]
       or: setup.py --help-commands
       or: setup.py cmd --help

    error: option --single-version-externally-managed not recognized
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\juanm\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\juanm\\AppData\\Local\\Temp\\pip-install-h7pab39z\\pythonnet_9cb3293501224b288fc1c9685bb36fbe\\setup.py'"'"'; __file__='"'"'C:\\Users\\juanm\\AppData\\Local\\Temp\\pip-install-h7pab39z\\pythonnet_9cb3293501224b288fc1c9685bb36fbe\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\juanm\AppData\Local\Temp\pip-record-gbzseqx0\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\juanm\appdata\local\programs\python\python39\Include\pythonnet' Check the logs for full command output.
```

But what I did was to install `wheel` with `pip`, and it worked at least for me.

What I did:
```
PS C:\Users\juanm\GitHub\Youngermaster\Raspberry-Pi-Codes\ZeroW\webview> pip install wheel
Collecting wheel
  Downloading wheel-0.36.2-py2.py3-none-any.whl (35 kB)
Installing collected packages: wheel
Successfully installed wheel-0.36.2
```

Then I tried again with the installation of the `pip package`:

```
PS C:\Users\juanm\GitHub\Youngermaster\Raspberry-Pi-Codes\ZeroW\webview> pip install pywebview
Collecting pywebview
  Using cached pywebview-3.4-py3-none-any.whl (325 kB)
Collecting pythonnet
  Using cached pythonnet-2.5.2.tar.gz (1.9 MB)
Requirement already satisfied: pycparser in c:\users\juanm\appdata\local\programs\python\python39\lib\site-packages (from pythonnet->pywebview) (2.20)
Building wheels for collected packages: pythonnet
  Building wheel for pythonnet (setup.py) ... done
  Created wheel for pythonnet: filename=pythonnet-2.5.2-cp39-cp39-win_amd64.whl size=79982 sha256=8e605aae2709705ed8149e38ceae981d95945f740714d75be3eac57a31b58f78
  Stored in directory: c:\users\juanm\appdata\local\pip\cache\wheels\2c\15\9c\554e578042cc422d36f134b642ed4e8c9a5223a54dd6cc8749
Successfully built pythonnet
Installing collected packages: pythonnet, pywebview
Successfully installed pythonnet-2.5.2 pywebview-3.4
```

So, if someone has this error, try installing `wheel` with `pip` (`pip install wheel`)

**Comment by mishraabhi0123** [comment]
I faced this issue as well. For some reason pip was not able to get the `.whl` for the `pythonnet`. This is how I resolved it.
1. Manually download the `.whl` file for `pythonnet` from [here](https://www.lfd.uci.edu/~gohlke/pythonlibs/#pythonnet) and save it to any location of your choice. Make sure to download according to your system specification.
2. If you are using virtual environment, activate the environment. Otherwise ignore this step.
3. Now run `pip install <path to pythonnet whl file>`
